### PR TITLE
Enable the use of EBS for data store

### PR DESCRIPTION
**Pull request by eherot**

### CODE REVIEW

File: attributes/elasticsearch.rb [hunk 1]
```diff
@@ -20,3 +20,4 @@
   'et_elk_elasticsearch_custom_config_cluster.name:' \
   "#{node['et_elk']['elasticsearch']['custom_config']['cluster.name']} AND " \
   'roles:es_master'
+default['et_elk']['storage_type'] = 'ebs'
```
Comment (jeffbyrnes) [inline attributes/elasticsearch.rb]: `default` or `set`? Isn’t this the top-level cookbook for this cluster?

Comment (eherot) [inline attributes/elasticsearch.rb]: This is the only place where a default for this value is going to be set. The storage cookbook doesn't look at `node['et_elk']['storage_type']` directly.

File: recipes/elasticsearch.rb [hunk 2]
```diff
@@ -21,16 +21,19 @@
...
 
+# The complicated logic of figuring out where we're going to put all this data
+if node['et_elk']['storage_type'] == 'ebs'
+  unless node['storage']['ebs_mounts']
+    fail 'Selected storage type is ebs but no mounts are available'
+  end
+  mounts = node['storage']['ebs_mounts']
+elsif node['storage']['ephemeral_mounts']
+  mounts = node['storage']['ephemeral_mounts']
+end
```
Comment (jeffbyrnes) [inline recipes/elasticsearch.rb]: What about:

```ruby
mounts = if node['et_elk']['storage_type'] == 'ebs'
           unless node['storage']['ebs_mounts']
             raise 'Selected storage type is ebs but no mounts are available'
           end
           
           node['storage']['ebs_mounts']
         elsif node['storage']['ephemeral_mounts']
           node['storage']['ephemeral_mounts']
         end
```

Fun fact: RuboCop changed its tune about `fail` v. `raise` a while back. Is your Sublime not bugging you about that?

File: recipes/elasticsearch.rb [hunk 3]
```diff
@@ -21,16 +21,19 @@
...
   allocated_memory "#{(node['memory']['total'].to_i * 0.4).floor / 1024}m"
-  if node['storage']['ephemeral_mounts'] &&
-     node['storage']['ephemeral_mounts'].any?
-    path_data(
-      package: node['storage']['ephemeral_mounts'].map do |mount|
-        "#{mount}/elasticsearch/data"
-      end.sort.join(',')
-    )
-  end
+  path_data(package: mounts.map { |mount| "#{mount}/elasticsearch/data" }.sort.join(',')) if mounts
```
Comment (jeffbyrnes) [inline recipes/elasticsearch.rb]: Maybe cleaner like so?

```ruby
path_data package: mounts.map { |mount| "#{mount}/elasticsearch/data" }.sort.join(',') if mounts
```

Hard to say; just throwing it out there.

**Comment by jeffbyrnes** [comment]
@eherot by the way, need to give Travis the `kitchen-sync` gem, or you get [this build failure](https://travis-ci.org/evertrue/et_elk-cookbook/builds/131501314#L305). I just fixed that on master, along with the RuboCop complaint.

Might be good to rebase this & see if it builds properly.

File: recipes/elasticsearch.rb (hunk 3 above)
Comment (eherot) [inline recipes/elasticsearch.rb]: This line doesn't work without the `()`.

Comment (jeffbyrnes) [inline recipes/elasticsearch.rb]: HugOps

File: recipes/elasticsearch.rb (hunk 2 above)
Comment (eherot) [inline recipes/elasticsearch.rb]: Ah yes good call.

**Comment by eherot** [comment]
`kitchen-sync` should be in here now.

**Comment by jeffbyrnes** [comment]
Looks like this is trying to install the HQ plugin, and failing.

**Comment by eherot** [comment]
Yes, that's fixed by changes in #24

**Comment by jeffbyrnes** [comment]
@eherot ah, ok. Well then, once that merges in, this should be good with a rebase.

**Pull request merged**